src/sentry/web/frontend/group_event_json.py

```diff
@@ -26,9 +29,13 @@ def get(self, request, organization, group_id, event_id_or_latest):
             # circumstances (such as a post_save signal failing)
             event = group.get_latest_event() or Event(group=group)
         else:
-            event = get_object_or_404(group.event_set, pk=event_id_or_latest)
+            if use_snuba:
+                event = SnubaEvent.objects.from_event_id(event_id_or_latest, group.project.id)
+            else:
+                event = get_object_or_404(group.event_set, pk=event_id_or_latest)
```

Comment:
Does this still cause problems when snuba is turned off? this seems like it won't find the event by hex`event_id` because we are explicitly looking for it in the  `pk` (postgres id) column

Reply:
Oh yeah i guess the switch for Snuba wasn't strictly necessary since we could've just supported both with `Event.objects.from_event_id()`. Updated to use SnubaEvent if the option is on anyway.

Reply:
Cool, yeah I guess technically now there is a slight behavior change in that you can now in theory grab an event that isn't actually in the group from the URL and it won't stop you... but I don't know if that's a huge deal or not.